test-loop-tests/src/tests/congestion_control.rs

```diff
@@ -59,7 +66,82 @@ fn slow_test_congestion_control_simple() {
 }
 
-fn setup(accounts: &Vec<AccountId>) -> (TestLoopEnv, AccountId) {
+#[cfg_attr(not(feature = "test_features"), ignore)]
+#[test]
+fn slow_test_one_shard_congested() {
+    init_test_logger();
+
+    let mut accounts = make_accounts(NUM_PRODUCERS + NUM_VALIDATORS);
+    let shard1_acc1: AccountId = "0000".parse().unwrap();
+    let shard1_acc2: AccountId = "1111".parse().unwrap();
+    let shard2_acc1: AccountId = "xxxx".parse().unwrap();
+    accounts.push(shard1_acc1.clone());
+    accounts.push(shard1_acc2.clone());
+    accounts.push(shard2_acc1.clone());
+
+    let runtime_config = RuntimeConfig::test();
+    let max_missed_chunks = runtime_config.congestion_control_config.max_congestion_missed_chunks;
+    let epoch_length = max_missed_chunks * 100;
+    let (env, rpc_id) = setup(&accounts, epoch_length, &["near"]);
+    assert_eq!(rpc_id, shard1_acc1);
+
+    // Assert correct shard layout
+    let shard_layout = &env.shared_state.genesis.config.shard_layout;
+    let shard1 = shard_layout.account_id_to_shard_id(&shard1_acc1);
+    let shard2 = shard_layout.account_id_to_shard_id(&shard2_acc1);
+    assert_eq!(shard1, shard_layout.account_id_to_shard_id(&shard1_acc2));
+    assert_ne!(shard1, shard2);
+
+    // Configure test env to drop chunks for shard 2
+    let dropped_chunks = [
+        (shard2, (0..epoch_length).map(|_| false).collect_vec()), // All chunks in the epoch
+    ]
+    .into_iter()
+    .collect();
+    let mut env = env.drop(DropCondition::ChunksProducedByHeight(dropped_chunks));
+
+    // Run for `max_missed_chunks * 2` blocks to make sure shard 2 is congested
+    env.run_blocks(max_missed_chunks * 2);
+
+    // Check if shard 2 is congested
+    let client_handle = env.node_datas[0].client_sender.actor_handle();
+    let client_actor = env.test_loop.data.get(&client_handle);
+    let head = client_actor.client.chain.get_head_block().unwrap();
+    let missed_chunks = head.block_congestion_info().get(&shard2).unwrap().missed_chunks_count;
+    assert!(missed_chunks >= max_missed_chunks);
+
+    // Send transfer from shard 1 to shard 1 – should succeed
+    let tx = prepare_transfer_tx(&mut env, &shard1_acc1, &shard1_acc2, ONE_NEAR);
+    let tx_outcome = execute_tx(
+        &mut env.test_loop,
+        &rpc_id,
+        TransactionRunner::new(tx, false),
+        &env.node_datas,
+        Duration::seconds(2),
+    )
+    .unwrap();
+    assert_matches!(tx_outcome.status, FinalExecutionStatus::SuccessValue(_));
+
+    // Send transfer from shard 1 to shard 2 – should fail, because shard 2 is congested
+    let tx = prepare_transfer_tx(&mut env, &shard1_acc1, &shard2_acc1, ONE_NEAR);
+    let tx_error = execute_tx(
+        &mut env.test_loop,
+        &rpc_id,
+        TransactionRunner::new(tx, false),
+        &env.node_datas,
+        Duration::seconds(2),
+    )
+    .unwrap_err();
+    assert_matches!(tx_error, InvalidTxError::ShardStuck { shard_id, .. } if shard_id == Into::<u32>::into(shard2));
```

Comment:
What's the `if` doing there? Is it making the assert conditional or making it stronger?

Reply:
It is making it stronger like this:
```rust
match tx_error {
  InvalidTxError::ShardStuck { shard_id, .. } if shard_id == Into::<u32>::into(shard2) => {},
  _ => panic!()
}
```